postprocessing/comstockpostproc/comstock_measure_comparison.py

```diff
@@ -51,16 +57,22 @@ def __init__(self, comstock_object, states, make_comparison_plots, make_timeseri
                 # filter dataset to upgrade and baseline only
                 up_base_id = '00'
                 upgrade_id = upgrade
-                if self.data.dtypes[self.UPGRADE_ID] == np.int64:
+                logger.info(f'self.data type is {type(self.data)}')
+                if self.data.select(self.UPGRADE_ID).dtypes == [pl.Int64]: # in test run it's pl.Int64
                     up_base_id = 0
                     upgrade_id = int(upgrade)
+                
 
                 # convert grouping column from cat to str to avoid processing errors with more than 2 measures
-                self.data[self.column_for_grouping] = self.data[self.column_for_grouping].astype(str)
+                self.data = self.data.with_columns(pl.col(self.UPGRADE_NAME).cast(str))
+                assert self.data.select(self.column_for_grouping).dtypes == [pl.String]
 
-                df_upgrade = self.data.loc[(self.data[self.UPGRADE_ID]==upgrade_id) | (self.data[self.UPGRADE_ID]==up_base_id), :]
+                # df_upgrade = self.data.loc[(self.data[self.UPGRADE_ID]==upgrade_id) | (self.data[self.UPGRADE_ID]==up_base_id), :]
+                df_upgrade = self.data.filter((pl.col(self.UPGRADE_ID) == upgrade_id) | (pl.col(self.UPGRADE_ID) == up_base_id))
 
                 color_map = {'Baseline': self.COLOR_COMSTOCK_BEFORE, upgrade_name: self.COLOR_COMSTOCK_AFTER}
+                df_upgrade = df_upgrade.collect().to_pandas()
```

Comment:
Can we really collect the dataframe at this point without first downselecting to only the columns actually used in the comparison plots? This needs to be tested with a full-sized dataset to make sure the memory usage is reasonable. Otherwise, it seems like it should be kept as a pl.LazyFrame and only collected inside the plotting functions.